plugins/modules/azure_rm_backupazurevm.py

```diff
@@ -0,0 +1,370 @@
+id:
+    description:
+        - VM backup protection details.
+    returned: always
+    type: str
+    sample: '{"response":{"id":"protection_id","name":"protection_item_name","properties":{}}}'
+'''
+
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_rest import GenericRestClient
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBaseExt
+import re
+import json
+import time
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    (NoAction, Create, Update, Delete) = range(4)
+
+
+class BackupAzureVM(AzureRMModuleBaseExt):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            recovery_vault_name=dict(
+                type='str',
+                required=True
+            ),
+            resource_id=dict(
+                type='str',
+                required=True
+            ),
+            backup_policy_id=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='create',
+                choices=['create', 'update', 'delete', 'stop', 'backup']
+            )
+        )
+
+        self.resource_group = None
+        self.recovery_vault_name = None
+        self.resource_id = None
+        self.backup_policy_id = None
+        self.state = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.url = None
+        self.status_code = [200, 201, 202, 204]
+        self.to_do = Actions.NoAction
+
+        self.body = {}
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = None
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        super(BackupAzureVM, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                            supports_check_mode=True,
+                                            supports_tags=True)
+
+    def get_api_version(self):
+        return '2019-05-13' if self.state == 'create' or self.state == 'update' or self.state == 'delete' or self.state == 'stop' else '2016-12-01'
+
+    def get_url(self):
+        if self.state == 'create' or self.state == 'update' or self.state == 'delete' or self.state == 'stop':
+            return '/subscriptions' + '/' + self.subscription_id \
+                   + '/resourceGroups' + '/' + self.resource_group + '/providers' \
+                   + '/Microsoft.RecoveryServices' + '/vaults' + '/' \
+                   + self.recovery_vault_name \
+                   + '/backupFabrics/Azure/protectionContainers/' \
+                   + 'iaasvmcontainer;iaasvmcontainerv2;' + self.resource_group \
```

Comment:
The string here is weird, why do you do this?

Reply:
Below is the API call for azure vm backup

PUT https://management.azure.com/Subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testVaultRG/providers/Microsoft.RecoveryServices/vaults/testVault/backupFabrics/Azure/protectionContainers/iaasvmcontainer;iaasvmcontainerv2;testRG;testVM/protectedItems/vm;iaasvmcontainerv2;testRG;testVM?api-version=2019-05-13


containerName = "iaasvmcontainer;"+{name}
protectedItemName = "vm;"+ {name}

eg.

containerName = "iaasvmcontainer;iaasvmcontainerv2;testRG;testVM"
protectedItemName = "vm;iaasvmcontainerv2;testRG;testVM

containrName and ProtectedItemName is combination of resouceGroup + VMName + Container string

Reply:
I see! Thank you very much！